packages/fxa-auth-server/lib/routes/utils/cms/localization.ts

```diff
@@ -188,35 +191,92 @@ export class CMSLocalization {
       ftlLines.push(`${entry.ftlId} = ${entry.value}`);
     }
 
-    return ftlLines.join('\n');
+    const ftlContent = ftlLines.join('\n');
+    return this.sanitizeFtlContent(ftlContent);
+  }
+
+  /**
+   * Sanitize entire FTL content to remove hidden Unicode characters
+   */
+  private sanitizeFtlContent(ftlContent: string): string {
+    return this.sanitizeContent(ftlContent, {
+      normalizeUnicode: false,
+      removeControlChars: false,
+      trimWhitespace: false,
+      normalizeLineEndings: true,
+      ensureFileTermination: true
+    });
   }
 
   /**
    * Sanitize content for FTL format
+   * @param content - The content to sanitize
+   * @param options - Sanitization options
    */
-  private sanitizeContent(content: string): string {
+  private sanitizeContent(content: string, options: {
+    normalizeUnicode?: boolean;
+    removeControlChars?: boolean;
+    trimWhitespace?: boolean;
+    normalizeLineEndings?: boolean;
+    ensureFileTermination?: boolean;
+  } = {}): string {
     if (!content || typeof content !== 'string') {
       return '';
     }
 
-    return content
-      // Normalize Unicode characters and remove problematic ones
-      .normalize('NFD') // Decompose Unicode characters
+    const {
+      normalizeUnicode = true,
+      removeControlChars = true,
+      trimWhitespace = true,
+      normalizeLineEndings = false,
+      ensureFileTermination = false
+    } = options;
+
+    let sanitized = content
+      .replace(/^\uFEFF/, '')
+      .replace(/[\u202A-\u202E\u2066-\u2069]/g, '')
+      .replace(/[\u200B-\u200D\uFEFF]/g, '')
+      .replace(/[\u2028\u2029]/g, normalizeLineEndings ? '\n' : '')
+      .replace(/[\u2060]/g, '');
+
+    // Apply optional sanitizations
+    if (normalizeUnicode) {
+      sanitized = sanitized.normalize('NFC');
+    }
+
+    if (removeControlChars) {
       // eslint-disable-next-line no-control-regex
-      .replace(/[\u0000-\u0008\u000B\u000C\u000E-\u001F\u007F-\u009F]/g, '') // Remove control characters (preserve tab, newline, carriage return)
-      .replace(/[\u200B-\u200D\uFEFF]/g, '') // Remove zero-width spaces and BOM
-      .trim(); // Remove leading/trailing whitespace
+      sanitized = sanitized.replace(/[\u0000-\u0008\u000B\u000C\u000E-\u001F\u007F-\u009F]/g, '');
+    }
+
+    if (normalizeLineEndings) {
+      sanitized = sanitized
+        .replace(/\r\n/g, '\n')
+        .replace(/\r/g, '\n');
+    }
+
+    if (trimWhitespace) {
+      sanitized = sanitized.trim();
+    }
+
+    if (ensureFileTermination) {
+      sanitized = sanitized.replace(/\n*$/, '\n');
+    }
+
+    return sanitized;
   }
 
   /**
-   * Generate FTL ID with l10nId prefix, component name, field name, and hash
+   * Generate FTL ID using only the hash of the string content
```

Comment:
polish: update comment to align with the change to include component hash + substring is 8 char not 12